mesonbuild/environment.py

```diff
@@ -257,6 +274,11 @@ def __init__(self, source_dir, build_dir, main_script_launcher, options, origina
             self.exe_suffix = 'exe'
             self.object_suffix = 'obj'
             self.win_libdir_layout = True
+        elif (not cross and mesonlib.is_cygwin()) \
+           or (cross and self.cross_info.has_host() and self.cross_info.config['host_machine']['system'] == 'cygwin'):
```

Comment:
This is actually under-indented (and the backslash above isn't necessary in parentheses).